scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModelPredictions.scala

```diff
@@ -0,0 +1,130 @@
+          BinomialPrediction3(
+            pred.classProbabilities(0),
+            pred.classProbabilities(1),
+            pred.calibratedClassProbabilities(0),
+            pred.calibratedClassProbabilities(1),
+            pred.contributions
+          )
+        }
+      } else {
+        udf[BinomialPrediction1, Row] { r: Row =>
+          val pred = easyPredictModelWrapper.predictBinomial(RowConverter.toH2ORowData(r))
+          BinomialPrediction1(
+            pred.classProbabilities(0),
+            pred.classProbabilities(1),
+            pred.calibratedClassProbabilities(0),
+            pred.calibratedClassProbabilities(1)
+          )
+        }
+      }
+    } else if (getCalculateContributions()) {
+      udf[BinomialPrediction2, Row] { r: Row =>
+        val pred = easyPredictModelWrapper.predictBinomial(RowConverter.toH2ORowData(r))
+        BinomialPrediction2(
+          pred.classProbabilities(0),
+          pred.classProbabilities(1),
+          pred.contributions
+        )
+      }
+    } else {
+      udf[BinomialPrediction0, Row] { r: Row =>
+        val pred = easyPredictModelWrapper.predictBinomial(RowConverter.toH2ORowData(r))
+        BinomialPrediction0(
+          pred.classProbabilities(0),
+          pred.classProbabilities(1)
+        )
+      }
+    }
+  }
+
+  def getBinomialPredictionSchema(): Seq[StructField] = {
+    val binomialSchemaBase = Seq("p0", "p1").map(StructField(_, DoubleType, nullable = false))
+
+    if (supportsCalibratedProbabilities()) {
+      val base = binomialSchemaBase ++ Seq("p0_calibrated", "p1_calibrated").map(StructField(_, DoubleType, nullable = false))
+      if (getCalculateContributions()) {
+        base ++ Seq(StructField("contributions", ArrayType(FloatType)))
+      } else {
+        base
+      }
+    } else if (getCalculateContributions()) {
+      binomialSchemaBase ++ Seq(StructField("contributions", ArrayType(FloatType)))
+    } else {
+      binomialSchemaBase
+    }
+  }
+
+  def getRegressionPredictionUDF(): UserDefinedFunction = {
+    if (getCalculateContributions()) {
+      udf[RegressionPrediction1, Row] { r: Row =>
+        val pred = easyPredictModelWrapper.predictRegression(RowConverter.toH2ORowData(r))
+        RegressionPrediction1(pred.value, pred.contributions)
+      }
+    } else {
+      udf[RegressionPrediction0, Row] { r: Row =>
+        val pred = easyPredictModelWrapper.predictRegression(RowConverter.toH2ORowData(r))
+        RegressionPrediction0(pred.value)
+      }
+    }
+  }
+
+  def getRegressionPredictionSchema(): Seq[StructField] = {
+    if (getCalculateContributions()) {
+      StructField("value", DoubleType) :: StructField("contributions", ArrayType(FloatType)) :: Nil
+    } else {
+      StructField("value", DoubleType) :: Nil
+    }
+  }
+}
+
+
+case class RegressionPrediction0(value: Double)
+
+case class RegressionPrediction1(value: Double, contributions: Array[Float])
```

Comment:
Would it be possible to name these case classes more descriptively? `RegressionPrediction1` -> `RegressionPredictionWithContributions`

Reply:
I had it like this, but changed as it was too long. But I have an idea, thanks!